utils/fitvids.test.js

```diff
@@ -0,0 +1,146 @@
+      width="720"
+      height="405">`
+  },
+  {
+    case: 'Link pasted into the Ghost editor and auto-wrapped in figure.kg-embed-card element',
+    html: `<figure class="kg-card kg-embed-card">
+        <iframe
+          width="200"
+          height="113"
+          src="https://www.youtube.com/embed/rfscVS0vtbw?feature=oembed"
+          frameborder="0"
+          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+          allowfullscreen=""
+          title="Embedded content"
+        >
+        </iframe>
+      </figure>`
+  },
+  {
+    case: 'Multiple embeds',
+    html: `<iframe
+        width="200"
+        height="113"
+        src="https://www.youtube.com/embed/rfscVS0vtbw?feature=oembed"
+        frameborder="0"
+        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
+        allowfullscreen=""
+        title="Embedded content"
+      ></iframe>
+      <iframe
+        src="https://player.vimeo.com/video/700486996?app_id=122963"
+        width="426"
+        height="226"
+        frameborder="0"
+        allow="autoplay; fullscreen; picture-in-picture"
+        allowfullscreen=""
+        title="Embedded content"
+      ></iframe>`
+  }
+];
+
+const checkEmbeddedVideos = obj => {
+  describe(obj.case, () => {
+    const dom = new JSDOM(obj.html);
+    const window = dom.window;
+    const document = window.document;
+    fitvids(window);
+
+    document.querySelectorAll('figure').forEach((figure, i) => {
+      const fluidWidthVideoContainer =
+        document.querySelectorAll('figure > div')[i];
+      const fluidWidthVideoWrapper =
+        document.querySelectorAll('figure > div > div')[i];
+      const embeddedVideo = document.querySelectorAll('figure > div > div > *')[
+        i
+      ];
+
+      test('there is no inner figure element', () => {
+        expect(figure.querySelectorAll('figure').length).toBe(0);
+      });
+
+      test('the figure element has the attributes and values', () => {
+        expect(figure.className).toBe('kg-card kg-embed-card');
+        expect(figure.getAttribute('data-test-label')).toBe('fitted');
+      });
+
+      test('the fluid width video container has the expected class name', () => {
+        expect(fluidWidthVideoContainer.className).toBe(
+          'fluid-width-video-container'
+        );
+      });
+
+      test('the fluid width video wrapper has the expected attributes and values', () => {
+        expect(fluidWidthVideoWrapper.className).toBe(
+          'fluid-width-video-wrapper'
+        );
+        expect(
+          parseFloat(fluidWidthVideoWrapper.style.paddingTop)
+        ).toBeTruthy();
+      });
+
+      test('the embedded video has the expected attributes and values', () => {
+        expect(
+          embeddedVideo.tagName === 'object' ? 'data' : 'src'
+        ).toBeTruthy();
```

Comment:
Thank you for catching this. I do this way too often when writing tests 🤦 

Now that I've thought about it more, `fitVids` doesn't do much to the embedded video element's original attributes. All it does is add `name="fitvidsn"`, and `width` and `height` if it doesn't exist to calculate the padding for one of the wrappers. Then it removes the `width` and `height` from the embedded video at the end of the script.

It seems like it might be better to just test for things that `fitVids` adds, and not to bother testing any of the original attributes that were already there. It would be possible to do a diff of the original existing embedded video attributes against the ones after running `fitVids`, but I think we can trust that the script won't mess with any of those.

Also, it seems like the videos are all responsive whether or not the `width` and `height` attributes are there in the final `fitVids` embedded videos. It may be better to remove the code from `fitVids` that removes those attributes, just in case someone views the page with CSS disabled for some reason.

Reply:
Yeah, it's easily done. I try and write the tests first to avoid that trap, but obviously that's not always possible.

Anyways, your refactors have made things much clearer, thanks!